code/game/machinery/Sleeper.dm

```diff
@@ -134,6 +134,33 @@
 		else
 			user << "<span class='warning'>\The [src] has a beaker already.</span>"
 		return
+	else if(istype(I, /obj/item/weapon/grab))
+
+		var/mob/living/L = I:affecting
+		visible_message("[user] starts putting [I:affecting] into the [src].", 3)
+
+		if (do_mob(user, I:affecting, 20, needhand = 0))
+			if(occupant)
+				user << "<span class='warning'>\The [src] is already occupied.</span>"
+				return
+			var/bucklestatus = L.bucklecheck(user)
+			if (!bucklestatus)//incase the patient got buckled during the delay
+				return
+			if (bucklestatus == 2)
+				var/obj/structure/LB = L.buckled
+				LB.user_unbuckle_mob(user)
+			if(!ismob(I:affecting))

```

Comment:
Instead of this check, you should most likely do:
```
if (L != G.affecting)
    return
```

This way, you ensure that it's the same mob you've been working with at the start.